### PR TITLE
Auto-generation of common reports

### PR DESCRIPTION
**Pull request by lchorbadjiev**
## Aim

We changed the default configuration of common reports to be enabled by default.

When someone tries to access a common report from `GPFInstance.get_common_report`, the method will check if the common report cache already exists, and if not, it will generate it.

The auto-generation of the common reports cache is added after loading studies when creating `GPFInstance`.

### CODE REVIEW

File: dae/dae/common_reports/common_report.py [hunk 1]
```diff
@@ -161,3 +163,42 @@ def to_dict(self, full=False):
...
+        if not os.path.exists(os.path.dirname(report_filename)):
+            os.makedirs(os.path.dirname(report_filename))
+        with open(report_filename, "w+", encoding="utf8") as crf:
+            json.dump(self.to_dict(full=True), crf)
+
+    @staticmethod
+    def load(report_filename: str) -> Optional[CommonReport]:
+        """Load a common report from a file.
+
+        I file does not exists returns None.
```
Comment (IvoTod) [inline dae/dae/common_reports/common_report.py:178]: typo, "if the file does not exist"

File: dae/dae/common_reports/common_report.py [hunk 2]
```diff
@@ -161,3 +163,42 @@ def to_dict(self, full=False):
...
+        with open(report_filename, "r", encoding="utf-8") as crf:
+            cr_json = json.load(crf)
+
+        return CommonReport(cr_json)
+
+    @staticmethod
+    def build_and_save(study, force=False):
+        """Build a common report for a study and saves it.
+
+        If the common reports and disabled for the study, the function skips
```
Comment (IvoTod) [inline dae/dae/common_reports/common_report.py:191]: typo, and/are

File: dae/dae/common_reports/common_report.py [hunk 3]
```diff
@@ -161,3 +163,42 @@ def to_dict(self, full=False):
...
+        building the report and returns None.
+
+        If the report already exists the default behavior is to skip building
+        the report and return None. You can force building the report by
+        passing `force=True` to the function.
+        """
+        if not study.config.common_report.enabled:
+            return None
+        if os.path.exists(study.config.common_report.file_path) and not force:
+            return None
```
Comment (IvoTod) [inline dae/dae/common_reports/common_report.py:201]: Seems counter intuitive to me to return None when a common report is already generated. Is this for a specific use case?

**Review by IvoTod — APPROVED** [review]

File: dae/dae/common_reports/common_report.py (hunk 1 above)
Comment (lchorbadjiev) [reply dae/dae/common_reports/common_report.py:178]: fixed

File: dae/dae/common_reports/common_report.py (hunk 2 above)
Comment (lchorbadjiev) [reply dae/dae/common_reports/common_report.py:191]: fixed

File: dae/dae/common_reports/common_report.py (hunk 3 above)
Comment (lchorbadjiev) [reply dae/dae/common_reports/common_report.py:201]: I changed it to load and return the report. This looks more consistent.

**Pull request merged**